### PR TITLE
Scientific calculator using JS

### PR DESCRIPTION
**Pull request by MS-0712**

### CODE REVIEW

File: index.html [hunk 1]
```diff
@@ -0,0 +1,240 @@
...
+          <tr>
+            <th colspan="5">
+              <input type="text" class="form-control mt-2 text-end" name="inp" id="inp"/>
+            </th>
+          </tr>
+          <tr>
+            <td colspan="5" class="p-0">
+              <table class="table table-borderless mb-0 mt-1">
+                <tr class="mx-0">
+                  <!-- <td class="p-0"><button class="btn text-dark" value="DEG" id="angleParam" onclick="degRad(this)">DEG</button></td> -->
```
Comment (rutvik-simformsolutions) [inline index.html]: remove unwanted code

File: index.html [hunk 2]
```diff
@@ -0,0 +1,240 @@
...
+            </td>
+          </tr>
+          <tr>
+            <td colspan="2" class="text-left text-light">
+              <img src="./img/trigo.png" height="20" width="20" alt="">
+              Trigonometry
+              <i onclick="toggleTrigo()" class="bi bi-caret-down-fill"></i>
+            </td>
+            <td colspan="2" class="text-light">
+              <img src="https://cdn0.iconfinder.com/data/icons/glyphpack/60/function-512.png" height="20" width="20" alt="">
```
Comment (rutvik-simformsolutions) [inline index.html]: provide value to alt attribute

File: index.html [hunk 3]
```diff
@@ -0,0 +1,240 @@
...
+                  </td>
+                  <td>
+                    <button class="btn trigo" onclick="updateInp(this)" value="sinh("> sinh </button>
+                  </td>
+                  <td>
+                    <button class="btn trigo" onclick="updateInp(this)" value="tanh("> tanh </button>
+                  </td>
+                </tr>
+                <!-- <tr class="mx-0">
+                  <td>
```
Comment (rutvik-simformsolutions) [inline index.html]: remove commented code

File: js/main.js [hunk 4]
```diff
@@ -0,0 +1,186 @@
...
+    { viewName: 'sinh', mathName: 'Math.sinh' },
+    { viewName: 'cosh', mathName: 'Math.cosh' },
+    { viewName: 'tanh', mathName: 'Math.tanh' },
+    { viewName: 'sin(', mathName: 'Math.sin(' },
+    { viewName: 'cos(', mathName: 'Math.cos(' },
+    { viewName: 'tan(', mathName: 'Math.tan(' },
+    { viewName: 'log', mathName: 'Math.log10' },
+    { viewName: 'ln', mathName: 'Math.log' },
+    { viewName: '√', mathName: 'Math.sqrt' }
+]
```
Comment (rutvik-simformsolutions) [inline js/main.js]: Missing semicolon.

File: js/main.js [hunk 5]
```diff
@@ -0,0 +1,186 @@
...
+
+let flag = false
```
Comment (rutvik-simformsolutions) [inline js/main.js]: Missing semicolon.
--

File: js/main.js [hunk 6]
```diff
@@ -0,0 +1,186 @@
...
+
+document.addEventListener('keydown', (event) => {
+    let key = event.key
```
Comment (rutvik-simformsolutions) [inline js/main.js]: Missing semicolon.
--

File: js/main.js [hunk 7]
```diff
@@ -0,0 +1,186 @@
...
+            if (/[0-9]π[0-9]/.test(expression))
+                expression = expression.replace(/π/, "*3.14*")
+            if (/π[0-9]/.test(expression))
+                expression = expression.replace(/π/, "3.14*")
+            if (/[0-8]π/.test(expression))
+                expression = expression.replace(/π/, "*3.14")
+            if (/π/.test(expression))
+                expression = expression.replace(/π/, "3.14")
+            if (/e[0-9]/.test(expression))
+                expression = expression.replace(/e/, "2.78*")
```
Comment (rutvik-simformsolutions) [inline js/main.js]: optimize this code using switch case

File: js/main.js [hunk 8]
```diff
@@ -0,0 +1,186 @@
...
+            if (/[0-8]e/.test(expression))
+                expression = expression.replace(/e/, "*2.78")
+            if (/e[^a-z]/.test(expression))
+                expression = expression.replace(/e/, "2.78")
+            console.log(expression);
```
Comment (rutvik-simformsolutions) [inline js/main.js]: remove console.log

File: js/main.js [hunk 9]
```diff
@@ -0,0 +1,186 @@
...
+        let ans = factorial(n)
+        let regex = new RegExp("(" + n + ")!");
+        return expression.replace(regex, ans)
+    }
+}
+
+function evalRes() {
+    let state = document.getElementById("inp")
+    let expression = state.value
+    if (expression != '') {
```
Comment (rutvik-simformsolutions) [inline js/main.js]: your try catch must at parent level

File: js/main.js [hunk 10]
```diff
@@ -0,0 +1,186 @@
...
+        try {
+            while (expression.includes('!'))
+                expression = factorialHandler(expression)
+
+            mathFunctions.forEach(element => {
+                if (expression.includes(element.viewName)) {
```
Comment (rutvik-simformsolutions) [inline js/main.js]: code is not in proper format

File: js/main.js [hunk 11]
```diff
@@ -0,0 +1,186 @@
...
+        let n = ''
+        for (let i = oc - 2; expression[i] != "("; i--)
+            n += expression[i];
+
+        n = n.split('').reverse().join('')
+        let ans = factorial(eval(n))
+        return expression.replace("(" + n + ")!", ans)
+    } else {
+        let n = ''
+        for (let i = oc - 1; i >= 0; i--) {
```
Comment (rutvik-simformsolutions) [inline js/main.js:146]: create a one loop and check if and else inside the loop it will decrease your LOC

File: js/main.js [hunk 12]
```diff
@@ -0,0 +1,186 @@
...
+        state.value = state.value.substr(0, state.value.length - 1)
+    else if (value == "N") {
+        let oc = expression.length, n = ''
+        for (let i = oc - 2; i >= 0; i--) {
+            if (/[0-9]/.test(expression[i]))
+                n += expression[i];
+            else
+                break
+        }
+        n = parseInt(n.split('').reverse().join(''), 10)
```
Comment (rutvik-simformsolutions) [inline js/main.js]: can you use switch case here for better code readability

**Review by rutvik-simformsolutions — CHANGES_REQUESTED** [review]

File: js/main.js (hunk 11 above)
Comment (MS-7121) [reply js/main.js:146]: Both the loops are handling different cases, so it's not possible to merge these loops into one and then apply if else conditions

**Review by rutvik-simformsolutions — APPROVED** [review]

**Pull request merged**